### PR TITLE
Fix Lints (Jannis und Ben)

### LINKED ISSUE #135: Refactor Class Comment Cache
- [x] Do a spike to decide if we need the cache
- [x] either remove the cache or actually use it

### PR DESCRIPTION
**Pull request by BenBals**
NB. Only merge after #125.

Fixes our lints in #136.

This pull request also removes the class comment cache and implements fuzzy matching for class comments while dramatically reducing the time the first algernon query after a reset takes.

Fixes #135.

This PR also includes @BenBals' refactorings around `do:`

### CODE REVIEW

**Comment by BenBals** [comment]
As far as I'm concerned this is ready for review. What do you think @sinnaj-r?

**Comment by BenBals** [comment]
@sinnaj-r could you take a look at the merge conflicts as they are probably because of your #139?

Note that you can use the "resolve conflicts" button right here in github.

File: packages/Algernon.package/ALGPackageType.class/class/algertemList.st [hunk 1]
```diff
@@ -1,10 +1,5 @@
...
 algertemList
-
-	| packages algertems |
-	algertems := OrderedCollection new.
-	packages := MCWorkingCopyBrowser new workingCopies.
-	packages do: [ :each | algertems add: (
-		ALGItem withAlgertype: (self with:each packageName))].
-	^ algertems
-
+
```
Comment (robertkirsten) [inline packages/Algernon.package/ALGPackageType.class/class/algertemList.st]: There should be no empty line here.

File: packages/AlgernonTests.package/ALGTestLifecycle.class/instance/numberOfResultListsFor..st [hunk 2]
```diff
@@ -1,4 +1,4 @@
 accessing
 numberOfResultListsFor: anAlgernon
```
Comment (robertkirsten) [inline packages/AlgernonTests.package/ALGTestLifecycle.class/instance/numberOfResultListsFor..st]: No empty line here too.

File: packages/Algernon.package/ALGCategoryType.class/class/algertemList.st [hunk 3]
```diff
@@ -1,7 +1,5 @@
 algertem lists
 algertemList
-	| classes algertems |
-	algertems := OrderedCollection new.
-	classes := SystemOrganization categories asOrderedCollection.
-	classes do: [ :each | algertems add: (ALGItem withAlgertype: (self with: each))].
-	^ algertems
\ No newline at end of file
+
```
Comment (robertkirsten) [inline packages/Algernon.package/ALGCategoryType.class/class/algertemList.st]: Empty line consistency.

**Review by robertkirsten — APPROVED** [review]
Threre are a few points where empty lines before returns aren't consistent, but other than that it looks very good.

**Comment by BenBals** [comment]
Thanks @robertkirsten, I fixed the whitespace issues!

**Pull request merged**